server.go

```diff
@@ -2254,9 +2254,30 @@ func (s *server) peerConnected(conn net.Conn, connReq *connmgr.ConnReq,
 	localFeatures.Set(lnwire.DataLossProtectOptional)
 	localFeatures.Set(lnwire.GossipQueriesOptional)
 
+	// We'll also construct a set of preferred feature bits, which we'll use
+	// to filter out peers if we make an outbound connection and have no
+	// channels with the node. This is useful to autopilot, and enforces
+	// strict policies on connections we make.
+	preferredRemoteFeatures := lnwire.NewRawFeatureVector()
+
+	// If we're in neutrino mode, we'll prefer those supporting data loss
+	// protection and gossip sync queries. This will help to establish
+	// connections with reliable and efficient peers.
+	if cfg.Bitcoin.Node == "neutrino" {
```

Comment:
This is a nice optimization, but I worry this can lead to unexpected behavior for some users. If they manually try to connect a node without these features, the connection will just fail, and they have no way of overriding that?

Instead I think we should apply the preferred features only for automatically created connections.